cli/dcoscli/main.py

```diff
@@ -60,8 +60,9 @@ def _main():
     if dcoscli.version != 'SNAPSHOT' and command and command != "config":
         try:
             cluster_id = mesos.DCOSClient().metadata().get('CLUSTER_ID')
-        except DCOSException:
-            raise
+        except DCOSException as e:
+            if "Authentication failed" in str(e):
```

Comment:
Should this be `... not in str(e)`?

Also, should we instead throw an specific `DCOSAuthException` and catch that instead? This exception can inherit `DCOSHTTPException`.